EIPS/eip-42.md

```diff
@@ -0,0 +1,399 @@
+    function isTrustedIssuer(address _issuer) external view returns(bool);
+    function getTrustedIssuerClaimTopics(IClaimIssuer _trustedIssuer) external view returns(uint[] memory);
+    function hasClaimTopic(address _issuer, uint _claimTopic) external view returns(bool);
+
+    // role setter
+    function transferOwnershipOnIssuersRegistryContract(address _newOwner) external;
+}
+```
+
+### Claim Topics Registry Interface
+
+The Claim Topics Registry stores all the trusted claim topics for the security token. The [ONCHAINID](https://tokeny.com/onchainid/) of token owners must contain claims of the claim topics stored in this smart contract. The ownership of this contract is given to the token issuer allowing them to manage this registry as per their requirements. <br>
+A detailed description of the functions can be found in the [interfaces folder](https://github.com/tokenyICO/eip-42/tree/main/interfaces) <br>
+
+```solidity
+interface IClaimTopicsRegistry {
+
+    // events
+    event ClaimTopicAdded(uint256 indexed claimTopic);
+    event ClaimTopicRemoved(uint256 indexed claimTopic);
+
+    // functions
+    // setters
+    function addClaimTopic(uint256 _claimTopic) external;
+    function removeClaimTopic(uint256 _claimTopic) external;
+
+    // getter
+    function getClaimTopics() external view returns (uint256[] memory);
+
+    // role setter
+    function transferOwnershipOnClaimTopicsRegistryContract(address _newOwner) external;
+}
+```
+
+## Test Cases
+
+The standard is implemented and tested with full coverage on Tokeny's [T-REX repository](https://github.com/TokenySolutions/T-REX)
+
+## Security Considerations
+
+The suite of Smart Contracts has been audited by an external and independent company. The results can be found in [this document](https://tokeny.com/wp-content/uploads/2020/05/Tokeny-Solutions_T-REX-v3_Smart-Contract-Audit-Report_Kapersky.pdf).
+
+## Notes
+
+### Total on-chain validation
+
+This standard is working completely on-chain. No off-chain validator is needed to perform transactions and token exchanges, thus introducing no external dependencies for the regulation of the exchanges.
+
+### KYC/AML Considerations
+
+Under some jurisdictions, a KYC/AML screening of one’s identity is necessary to hold tokens representing an underlying financial instrument. This standard relies heavily on identity contracts and validation through claims, enforcing the screening of identities before being able to hold tokens.
+
+### Cap-table immutability
+Through heavy identity integration, the standard is able to ensure the immutability of the cap-table on the blockchain. Since tokens are linked to identities instead of wallets, it is always possible to trace to whom the token belongs to, even in the case of lost wallets.
+
+## References
+
+- [ONCHAINID documentation](https://docs.onchainid.com/)
+- [T-REX White Paper](https://tokeny.com/wp-content/uploads/2019/12/Whitepaper-T-REX-Security-tokens.pdf)
```

Comment:
Please be sure to include the copyright notice at the bottom as listed in the EIP template.